### PR TITLE
feat: react18 legacy mode & react17 support

### PR DESCRIPTION
**Pull request by jesse23**
https://github.com/Daydreamer-riri/vite-react-ssg/issues/49

### CODE REVIEW

**Comment by Daydreamer-riri** [comment]
Thanks! I think we can handle the rendering function internally through the react-dom exported version instead of passing `useReact17` in manually by the user: https://github.com/facebook/react/blob/4ff5f5719b348d9d8db14aaa49a48532defb4ab7/packages/react-dom/src/client/ReactDOM.js#L182

Like:
```ts
const isReact18 = Number((version || '').split('.')[0]) > 17;
```

**Comment by jesse23** [comment]
@Daydreamer-riri there is a use case for us, we need to use it with React18 with 'Old API', which is a valid case...that is why option is needed.

**Comment by jesse23** [comment]
comments addressed 🚀

**Comment by Daydreamer-riri** [comment]
Thank you for your contribution! It's time for me to go to bed, and I'll review the changes as soon as possible tomorrow. Looking forward to the merger!

**Comment by Daydreamer-riri** [comment]
I left a comment. You can check it out whenever you are free!

**Comment by jesse23** [comment]
@Daydreamer-riri  my bad where is your comments? I scanned the diff and not seeing any...

If it is below, I have changed the boolean to `useLegacyRender` as discussed
https://github.com/Daydreamer-riri/vite-react-ssg/pull/50#issuecomment-2537872356

**Review by Daydreamer-riri — CHANGES_REQUESTED** [review]

**Comment by Daydreamer-riri** [comment]
Sorry, I forgot to submit it.

File: README.md [hunk 1]
```diff
@@ -609,6 +609,11 @@ export default defineConfig({
 })
 ```
 
+### React17 Support
+
+- for react18, with flag `useLegacyRender: true`, it will use the legacy `render` and `hydrate` methods.
+- for react17, on top of above, you will need minor update to react and react-dom [example](https://github.com/jesse23/webpack-test-bed/blob/main/scripts/define-react-exports.js) to polyfill the mjs import and the `react-dom/client`.
```
[the remark this answers is not in the document]
Comment (jesse23) [reply README.md:615]: OK I don't want to hack the repo a lot is my former intention...but let me try to port that and see

[the remark this answers is not in the document]
Comment (Daydreamer-riri) [reply README.md:615]: Haha, I understand. I prefer to make a feature as complete as possible if we're going to support it. If you need any help, please feel free to let me know, and I will do my best to help.

[the remark this answers is not in the document]
Comment (jesse23) [reply README.md:615]: Yes I'll try to adopt the `import { createRoot } from 'react-dom'` to make it minimum. Will try to have a commit soon

[the remark this answers is not in the document]
Comment (jesse23) [reply README.md:615]: @Daydreamer-riri I made a simplified version of react-helper, please take a look. I have tested for both r17, r18 compatible and r18

**Review by Daydreamer-riri — APPROVED** [review]

**Pull request merged**